src/rules/no-t-call-in-react-component.ts

```diff
@@ -0,0 +1,314 @@
+import { ParserServices, TSESTree } from '@typescript-eslint/utils'
+import { createRule } from '../create-rule'
+import { RuleContext, SourceCode } from '@typescript-eslint/utils/ts-eslint'
+import ts, { type Expression, type SourceFile, type TypeChecker } from 'typescript'
+
+type Node = TSESTree.Node
+const { AST_NODE_TYPES } = TSESTree
+export const name = 'no-t-call-in-react-component'
+export const rule = createRule({
+  name,
+  meta: {
+    docs: {
+      description: 'use _(msg`...`) instead of t`...` and t() in react components',
+      recommended: 'error',
+    },
+    messages: {
+      default:
+        't`...` and t() call should not be used in react components, use _(msg`...`) instead',
+      fix1: 'Replace with _(msg`...`)',
+      fix2: 'Replace with <Trans>...</Trans>',
+    },
+    fixable: 'code',
+    hasSuggestions: true,
+    schema: [
+      {
+        type: 'object',
+        properties: {},
+        additionalProperties: false,
+      },
+    ],
+    type: 'problem' as const,
+  },
+
+  defaultOptions: [],
+
+  create: (context) => {
+    return {
+      'TaggedTemplateExpression[tag.name=t]'(node: TSESTree.TaggedTemplateExpression) {
```

Comment:
This query is not complete. There are different signatures of `t` macro which should be also covered by this rule. Check `LinguiTaggedTemplateExpressionMessageQuery` versus `LinguiCallExpressionMessageQuery` in the `helpers.ts`